pkg/sharing/placeholder_secret_test.go

```diff
@@ -52,6 +52,46 @@ func Test_SecretReconciler_respectsNamespaces(t *testing.T) {
 		assert.Equal(t, sourceSecret.Data[".dockerconfigjson"], placeholderSecret2.Data[".dockerconfigjson"])
 	})
 
+	t.Run("star export skips annotated namespaces", func(t *testing.T) {
+		sourceSecret, placeholderSecret1, placeholderSecret2 := resourcesUnderTest()
+		excludedNs := corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        placeholderSecret2.Namespace,
+				Annotations: map[string]string{"secretgen.carvel.dev/excluded-from-wildcard-matching": ""},
+			},
+		}
+		includedNs := corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: placeholderSecret1.Namespace,
+			},
+		}
+		secretExport := secretExportFor(sourceSecret, "*")
+		secretExportReconciler, secretReconciler, k8sClient := placeholderReconcilers(&sourceSecret, &placeholderSecret1, &placeholderSecret2, &secretExport, &excludedNs, &includedNs)
+		reconcileObject(t, secretExportReconciler, &secretExport)
+		reconcileObject(t, secretReconciler, &placeholderSecret1)
+		reconcileObject(t, secretReconciler, &placeholderSecret2)
+		// placeholder secret2 should have its original contents for auths and a helpful status message
+		originalPlaceholder2Data := append([]byte{}, placeholderSecret2.Data[".dockerconfigjson"]...)
```

Comment:
i dont get why we are capturing `originalPlaceholder2Data` _after_ doing reconciles. wouldnt moving it above all the reconciler stuff make it more clear that it's not being modified by all the reconcile*

Reply:
oh, sure, i could move it -- might make it easier to skim but at the cost of obscuring what's actually happening:  even though you pass a pointer to the reconciler calls, they don't actually modify the input arg - eventually somebody makes a copy.

  that's why we have to reload at all - if the reconcilers were modifying the structs that we have references to we wouldn't need to reload!

Reply:
see e.g. line 83 below this i tried to call out:  after we add the annotation we actually have to remake the fake-client because otherwise it'll keep returning (a copy of) the same struct that i passed in originally.

Reply:
I added more comments trying to explain...